packages/flutter/lib/src/material/bottom_sheet_theme.dart

```diff
@@ -0,0 +1,104 @@
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+/// Defines default property values for [BottomSheet] widgets.
+///
+/// Descendant widgets obtain the current [BottomSheetThemeData] object
+/// using `Theme.of(context).bottomSheetTheme`. Instances of
+/// [BottomSheetThemeData] can be customized with
+/// [BottomSheetThemeData.copyWith].
+///
+/// Typically a [BottomSheetThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.bottomSheetTheme].
+///
+/// All [BottomSheetThemeData] properties are `null` by default.
+/// When null, the [BottomSheet] will provide its own defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+class BottomSheetThemeData extends Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.bottomSheetTheme].
+  const BottomSheetThemeData({
+    this.backgroundColor,
+    this.elevation,
+    this.shape,
+  });
+
+  /// Color to be used for the [BottomSheet]'s background.
+  final Color backgroundColor;
+
+  /// The z-coordinate to be used for the [BottomSheet]'s elevation.
+  final double elevation;
+
+  /// The shape to be used for the bottom sheet's [Material].
+  final ShapeBorder shape;
+
+  /// Creates a copy of this object with the given fields replaced with the
+  /// new values.
+  BottomSheetThemeData copyWith({
+    Color backgroundColor,
+    double elevation,
+    ShapeBorder shape,
+  }) {
+    return BottomSheetThemeData(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      elevation: elevation ?? this.elevation,
+      shape: shape ?? this.shape,
+    );
+  }
+
+  /// Linearly interpolate between two bottom sheet themes.
+  ///
+  /// If both arguments are null then null is returned.
+  ///
+  /// {@macro dart.ui.shadow.lerp}
+  static BottomSheetThemeData lerp(BottomSheetThemeData a, BottomSheetThemeData b, double t) {
+    assert(t != null);
+    if (a == null && b == null)
+      return null;
+    return BottomSheetThemeData(
+      backgroundColor: Color.lerp(a?.backgroundColor, b?.backgroundColor, t),
+      elevation: lerpDouble(a?.elevation, b?.elevation, t),
+      shape: ShapeBorder.lerp(a?.shape, b?.shape, t),
+    );
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      backgroundColor,
+      elevation,
+      shape,
+    );
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final BottomSheetThemeData otherData = other;
+    return otherData.backgroundColor == backgroundColor
+        && otherData.elevation == elevation
+        && otherData.shape == shape;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    const BottomSheetThemeData defaultData = BottomSheetThemeData();
```

Comment:
I think this is unnecessary, since all of the fields in this class default to `null`, you can probably just pass `null` directly into the `defaultValue: ` args below.

Reply:
Done